src/plugins/intel_gna/src/transformations/matops_to_dwsc.cpp

```diff
@@ -0,0 +1,261 @@
+
+    if (input_shape.size() == 0)
+        return false;
+
+    uint64_t N, C, H, W;
+    switch (input_shape.size()) {
+    case 4:
+        N = input_shape[0];
+        C = input_shape[1];
+        H = input_shape[2];
+        W = input_shape[3];
+        break;
+    case 2:
+        N = 1;
+        C = input_shape[1];
+        H = 1;
+        W = input_shape[0];
+        break;
+    default:
+        return false;
+    }
+
+    uint64_t N_params, C_params, H_params, W_params;
+    switch (params_shape.size()) {
+    case 4:
+        N_params = params_shape[0];
+        C_params = params_shape[1];
+        H_params = input_shape[2];
+        W_params = input_shape[3];
+        break;
+    case 2:
+        N_params = params_shape[0];
+        C_params = params_shape[1];
+        H_params = 1;
+        W_params = 1;
+        break;
+    case 1:
+        N_params = 1;
+        C_params = params_shape[0];
+        H_params = 1;
+        W_params = 1;
+        break;
+    default:
+        return false;
+    }
+
+    // getting the parent(previous) node
+    const Output<Node>& parent = math_node->input_value(0);
+    if (nullptr != std::dynamic_pointer_cast<ngraph::opset1::Multiply>(math_node) && IsFusable(parent))
+        return false;
+
+    auto G = C;             // number of groups
+    auto Co = (uint64_t)1;  // no. kernels in a group
+    auto Ci = (uint64_t)1;  // no. input channels in a group
+    auto Kh = (uint64_t)1;  // kernel height
+    auto Kw = (uint64_t)1;  // kernel width
+
+    // Initialize dwsc (group convolution) kernel weights as 1s
+    std::vector<float> dwsc_weights(Kh * Kw * G, 0.0f);
+    float* dwsc_weight_ptr = dwsc_weights.data();
+    if (nullptr == std::dynamic_pointer_cast<ngraph::opset1::Multiply>(math_node)) {
+        for (uint32_t i = 0; i < Kh * Kw * G; i++)
+            *(dwsc_weight_ptr + i) = 1;
+    } else {
+        auto weights_const =
+            std::dynamic_pointer_cast<ngraph::opset1::Constant>(math_node->input_value(1).get_node_shared_ptr());
+        const float* weights_ptr = weights_const->get_data_ptr<float>();
+        for (uint32_t i = 0; i < Kh * Kw * G; i++)
+            *(dwsc_weight_ptr + i) = *(weights_ptr + i);
+    }
+
+    // Create a constant vector of weights
+    auto dwsc_weights_const =
+        ngraph::op::Constant::create(ngraph::element::f32, Shape{G, Co, Ci, Kh, Kw}, dwsc_weights);
+    dwsc_weights_const->set_friendly_name("dwsc_weights");
+
+    // Initializing group convolution parameters
+    const Strides& strides = Strides({1, 1});
+    const CoordinateDiff& pads_begin = CoordinateDiff({0, 0});
+    const CoordinateDiff& pads_end = CoordinateDiff({0, 0});
+    const Strides& dilations = Strides({1, 1});
+    const ngraph::op::PadType& auto_pad = ngraph::op::PadType::EXPLICIT;
+
+    // Initializing the dwsc node
+    std::shared_ptr<ov::op::v1::GroupConvolution> new_dwsc;
+    if (4 != input_shape.size()) {
+        std::shared_ptr<ov::op::v1::Reshape> input_4d;
```

Comment:
1. just wonder if creation of reshape would be moved to separate method would increase readability of the code.
2. assigning 8 and 65528 to some constant with valuable name would help understand meaning of this values.